### PR TITLE
Move check for login and email set by automation to AutomationHelper

### PR DESCRIPTION
**Pull request by daehn**
**What's in this PR?**
* `AutomationHelper` is now a singleton 🙊 and checks the NSProcessInfo arguments.
* Moved the check for username and email provided via the command line from `Settings` to `AutomationHelper`.
* Removed the separate command line flag to skip first time login alerts and set `skipFirstLoginAlerts ` to `true` as soon as username and email are provided via the command line.
* In order to disable autocorrection the ~~`--disableautocorrection`~~ `--disable-autocorrection` argument can be provided via the command line.

### CODE REVIEW

File: WireExtensionComponents/Utilities/AutomationHelper.swift [hunk 1]
```diff
@@ -18,27 +18,70 @@
...
+        super.init()
+        checkCommandLineArguments()
     }
-
+    
+    private enum AutomationKey: String {
+        case Email = "--loginemail="
+        case Password = "--loginpassword="
+        case LogNetwork = "--debug-log-network"
+        case DisableAutocorrection = "--disableautocorrection"
```
Comment (typfel) [inline WireExtensionComponents/Utilities/AutomationHelper.swift]: Would look nicer if we are consistent with the - for spaces like `--disable-auto-correction`

Comment (daehn) [inline WireExtensionComponents/Utilities/AutomationHelper.swift]: @typfel I changed `--disableautocorrection` to `--disable-autocorrection` in 8b8f384, but I didn't want to touch the email and password arguments as they are already used by automation.

File: WireExtensionComponents/Utilities/AutomationHelper.swift [hunk 2]
```diff
@@ -18,27 +18,70 @@
...
+    ///  - returns: `true` if values for `--loginemail=` and --`loginpassword=` have been provided on the command line
+    private(set) var skipFirstLoginAlerts = false
+    
+    ///  - returns: The `ZMEmailCredentials` specified with the `--loginemail=` and --`loginpassword=` arguments on the command line
+    private(set) var automationEmailCredentials: ZMEmailCredentials? = nil
+    
     ///  - returns: The value specified for the `DisableAutoCorrection` argument on the command line
-    class var disableAutoCorrection: Bool {
-        return defaults.boolForKey("DisableAutoCorrection")
+    private(set) var disableAutoCorrection = false
```
Comment (typfel) [inline WireExtensionComponents/Utilities/AutomationHelper.swift]: For even more consistency this could be renamed to `disableAutocorrection `

**Comment by daehn** [comment]
@typfel please-re-review

**Comment by daehn** [comment]
@typfel merged back develop with the test fixes

**Pull request merged**